torch/_inductor/fx_passes/post_grad.py

```diff
@@ -795,8 +795,16 @@ def can_inplace(node, mutated_arg):
                     graph.erase_node(copy_node)
 
                 node.target = inplaceable_op.inplace_op
-    for node, replacement in replace_list:
-        node.replace_all_uses_with(replacement)
+
+    def get_replacement(node):
+        ori_node = node
+        while node in replace_dict:
+            node = replace_dict[node]
```

Comment:
@oulgen To make sure: we can't have cycles in the `replace_dict`, right? I assume it is a DAG.

Reply:
nope, it would require fx graph to have a cycle which is not possible